middleware.go

```diff
@@ -0,0 +1,640 @@
+// Copyright © 2024 Meroxa, Inc.
```

Comment:
I'm not against this suggestion, but I think it might be a bit early for this. We currently only have 2 middleware types and they go together hand in hand, so even if we were to extract them, I'd like to keep them together in a file like `middleware_schema.go`. Furthermore, I currently don't anticipate us adding any more middleware, not in the short term at least, so I suggest leaving them in this file for now and cycling back to this if/when we add more. WDYT?

Reply:
@lovromazgon I personally still think that splitting this into different files could be good for visibility or maintainability in the future 🤷 .

Reply:
We discussed this offline and decided to keep it in `middleware.go` for now, also to keep it consistent with the connector SDK middleware. We may want to revisit this when we add more middleware though.